tests/unit/via/services/youtube_test.py

```diff
@@ -0,0 +1,33 @@
+from unittest.mock import sentinel
+
+import pytest
+
+from via.services.youtube import YoutubeService, factory
+
+
+class TestYoutubeService:
+    @pytest.mark.parametrize(
+        "url,video_id",
+        [
+            ("not_an_url", None),
+            ("https://notyoutube:1000000", None),
+            ("https://notyoutube.com", None),
+            ("https://youtube.com", None),
+            ("https://youtube.com?param=nope", None),
+            ("https://youtube.com?v=", None),
+            ("https://www.youtube.com/watch?v=VIDEO_ID", "VIDEO_ID"),
+            ("https://www.youtube.com/watch?v=VIDEO_ID&t=14s", "VIDEO_ID"),
```

Comment:
Support more formats see:

https://stackoverflow.com/a/8260383/15031259

And also not there:

https://www.youtube.com/live/4Lb2JoDN70o?feature=share
https://www.youtube.com/shorts/ktl1ncCTBGg

Reply:
I think I'm probably going to have LMS normalize the YouTube URLs into some custom format (something like: `video://youtube/{id}` before sending them to Via's `/video?url={url}` view. So the `/video` view won't need to support real YouTube URLs, so I think I'll probably end up removing this real YouTube URL parsing from Via (and putting it into LMS instead).

We would still need the real YouTube URL parsing in Via if we ever wanted to add YouTube support to public Via: the root `/<REAL_YOUTUBE_URL>` view would have to parse and normalize real YouTube URLs in the same way that LMS is going to and forward them to `/video?url={normalized_youtube_url}`.

Anyway, I think we *will* need this real YouTube URL parsing code at least in LMS even if we don't add YouTube support to public Via, so I might as well add support for more formats here now and it'll get cut-pasted into LMS later..